core/save-profiledata.cpp

```diff
@@ -175,32 +187,214 @@ static std::string format_st_event(const plot_data &entry, const plot_data &next
+		replace_all(format_string, "[ead]", format_string_std("%02.2f %s", value, unit));
+	} else {
+		replace_all(format_string, "[ead]", "");	
+	}
+
+	if (entry.end > 0) {
+		value = get_depth_units(entry.end, &decimals, &unit);
+		replace_all(format_string, "[end]", format_string_std("%02.2f %s", value, unit));
+	} else {
+		replace_all(format_string, "[end]", "");	
+	}
+	
+	if (entry.eadd > 0) {
+		value = get_depth_units(entry.eadd, &decimals, &unit);
+		replace_all(format_string, "[eadd]", format_string_std("%02.2f %s", value, unit));
+	} else {
+		replace_all(format_string, "[eadd]", "");	
+	}
+	
+	value = get_vertical_speed_units(entry.speed, &decimals, &unit);
+	if (entry.speed > 0)
+		/* Ascending speeds are positive, descending are negative */
+		value *= -1;
+	replace_all(format_string, "[speed]", format_string_std("%02.2f %s", value, unit));
+	
+	if (entry.in_deco) {
+		replace_all(format_string, "[in_deco]", translate("gettextFromC", "In deco"));
+	} else {
+		replace_all(format_string, "[in_deco]", "");	
+	}
+
+	if (entry.ndl_calc > 0) {
+		if (entry.ndl_calc < 7200) {
+			replace_all(format_string,"[ndl_calc]", format_string_std("%d:%02d", FRACTION_TUPLE(entry.ndl_calc, 60)));
+		} else {
+			replace_all(format_string,"[ndl_calc]", ">2h");
+		}
+	} else {
+		replace_all(format_string, "[ndl_calc]", "");	
+	}
+
+	if (entry.tts_calc > 0) {
+		replace_all(format_string,"[tts_calc]", format_string_std("%d:%02d", FRACTION_TUPLE(entry.tts_calc, 60)));
+	} else {
+		replace_all(format_string, "[tts_calc]", "");	
+	}
+
+	if (entry.stoptime_calc > 0) {
+		replace_all(format_string,"[stoptime_calc]", format_string_std("%d:%02d", FRACTION_TUPLE(entry.stoptime_calc, 60)));
+	} else {
+		replace_all(format_string, "[stoptime_calc]", "");	
+	}
+	
+	if (entry.stopdepth_calc > 0) {
+		value = get_depth_units(entry.stopdepth_calc, &decimals, &unit);
+		replace_all(format_string, "[stopdepth_calc]", format_string_std("%02.2f %s", value, unit));
+	} else {
+		replace_all(format_string, "[stopdepth_calc]", "");	
+	}
+	
+	if (entry.heartbeat > 0) {
+		replace_all(format_string, "[heartrate]", format_string_std("%d", entry.heartbeat));
+	} else {
+		replace_all(format_string, "[heartrate]", "");	
+	}
+	
 	if (entry.surface_gf > 0.0) {
-		res += format_string_std("sGF=%.1f%% ", entry.surface_gf);
+		replace_all(format_string, "[surface_gf]", format_string_std("%.1f%%", entry.surface_gf));
+	} else {
+		replace_all(format_string, "[surface_gf]", "");	
 	}
+
+	if (entry.current_gf > 0.0) {
+		replace_all(format_string, "[current_gf]", format_string_std("%.1f%%", entry.current_gf));
+	} else {
+		replace_all(format_string, "[current_gf]", "");	
+	}
+
+	if (entry.density > 0) {
+		replace_all(format_string, "[density]", format_string_std("%.1fg/ℓ", entry.density));
+	} else {
+		replace_all(format_string, "[density]", "");	
+	}
+
+	if (entry.icd_warning) {
+		replace_all(format_string, "[icd_warning]", translate("gettextFromC", "ICD in leading tissue"));
```

Comment:
Seems to be a bit wordy, I am worried that the subtitles will be displayed in a very small font if this is ever hit during a dive.

Reply:
I agree that it may be a bit verbose. Something like "ICD warning" might be better. However, "ICD in leading tissue" is used in the plot, so I don't know if we want to keep the same formulation for consistency?